lib/doctest.js

```diff
@@ -197,20 +212,46 @@ function processLine(
     } else if (trimmedLine === options.closingDelimiter) {
       accum.state = CLOSED;
     } else if (trimmedLine.charAt (0) === '>') {
+      accum.state = INPUT;
       value = stripLeading (1, ' ', stripLeading (1, '>', trimmedLine));
       accum.tests.push ($test = {});
-      $test[accum.state = INPUT] = {value: value};
+      $test[INPUT] = {value: value};
+      $test[OUTPUT] = {values: []};
       input ($test);
-    } else if (trimmedLine.charAt (0) === '.') {
+    } else if (accum.state === INPUT && trimmedLine.charAt (0) === '.') {
+      value = stripLeading (1, ' ', stripLeading (Infinity, '.', trimmedLine));
+      $test = accum.tests[accum.tests.length - 1];
+      $test[INPUT].value += '\n' + value;
+      appendToInput ($test);
+    } else if ((accum.state === OUTPUT || accum.state === LOG) &&
+               (trimmedLine.charAt (0) === '.')) {
       value = stripLeading (1, ' ', stripLeading (Infinity, '.', trimmedLine));
       $test = accum.tests[accum.tests.length - 1];
-      $test[accum.state].value += '\n' + value;
-      (accum.state === INPUT ? appendToInput : appendToOutput) ($test);
-    } else if (accum.state === INPUT) {
+      $test[OUTPUT].values[$test[OUTPUT].values.length - 1].value += '\n'
+                                                                  + value;
+      appendToOutput ($test);
+    } else if ((match = MATCH_LOG.exec (trimmedLine)) != null) {
+      accum.state = LOG;
+      value = stripLeading (1, ' ', trimmedLine.replace (MATCH_LOG, ''));
```

Comment:
We could extract the remainder from `match`:

```diff
--- a/lib/doctest.js
+++ b/lib/doctest.js
@@ -181,4 +181,2 @@ var LOG = 'log';

-var MATCH_LOG = /^\[([a-zA-Z]+)\]:/;
-
 //  normalizeTest :: { output :: { value :: String } } -> Undefined
@@ -232,5 +230,5 @@ function processLine(
       appendToOutput ($test);
-    } else if ((match = MATCH_LOG.exec (trimmedLine)) != null) {
+    } else if ((match = /^\[([A-Za-z]+)\]:(.*)$/.exec (trimmedLine)) != null) {
       accum.state = LOG;
-      value = stripLeading (1, ' ', trimmedLine.replace (MATCH_LOG, ''));
+      value = stripLeading (1, ' ', match[2]);
       $test = accum.tests[accum.tests.length - 1];
```

Why do we restrict channel names to `[A-Za-z]+`?